flink-runtime-web/web-dashboard/src/app/pages/job/checkpoints/job-checkpoints.component.ts

```diff
@@ -37,24 +40,52 @@ export class JobCheckpointsComponent implements OnInit {
 
   public moreDetailsPanel = { active: false, disabled: false };
 
-  constructor(private readonly jobService: JobService, private readonly cdr: ChangeDetectorRef) {}
+  private refresh$ = new Subject<void>();
+  private destroy$ = new Subject<void>();
+
+  constructor(
+    private readonly jobService: JobService,
+    private readonly jobLocalService: JobLocalService,
+    private readonly cdr: ChangeDetectorRef
+  ) {}
 
   public ngOnInit(): void {
-    this.jobService.jobDetail$.pipe(distinctUntilChanged((pre, next) => pre.jid === next.jid)).subscribe(data => {
-      this.jobDetail = data;
-      this.jobService.loadCheckpointStats(this.jobDetail.jid).subscribe(stats => {
-        this.checkPointStats = stats;
+    this.jobLocalService
+      .jobDetailChanges()
+      .pipe(
+        distinctUntilChanged((pre, next) => pre.jid === next.jid),
+        takeUntil(this.destroy$)
+      )
+      .subscribe(data => {
+        this.jobDetail = data;
         this.cdr.markForCheck();
+        this.refresh$.next();
       });
-      this.jobService.loadCheckpointConfig(this.jobDetail.jid).subscribe(config => {
+
+    this.refresh$
+      .pipe(
+        switchMap(() =>
+          forkJoin([
+            this.jobService.loadCheckpointStats(this.jobDetail.jid),
+            this.jobService.loadCheckpointConfig(this.jobDetail.jid)
```

Comment:
We should add some `catchError` handling to these two requests, otherwise any request error will kill the entire refresh pipe.